codeforamerica/vita-min
# Nj 67 disabled exemption

## Link to pivotal/JIRA issue
- https://github.com/newjersey/affordability-pm/issues/67#issue-2553710897

> Acceptance criteria

>GIVEN I have gotten past the screener and the directfile import AND I have not claimed the blind exemption on my federal return
WHEN I am answering NJ specific questions
THEN I see radio buttons that ask me if I was disabled on the last day of the tax year

>GIVEN I have gotten past the screener and the directfile import and I am filing jointly with my spouse/CU partner AND my spouse has not claimed the blind exemption on our federal return
WHEN I am answering NJ specific questions
THEN I see radio buttons that ask me if my spouse was disabled on the last day of the tax year

>GIVEN I have gotten past the screen and the directfile import
WHEN I am answering NJ specific questions
THEN I see helper text explaining the definition of disabled

>GIVEN I am about to submit my state file data
WHEN I see the page that allows me to review my answers
THEN I can go back and edit my disability status

>GIVEN I am about to submit my state file data and I am filing jointly
WHEN I see the page that allows me to review my answers
THEN I can go back and edit my partner's disability status

>GIVEN I have submitted my state return
WHEN I download the PDF
THEN it shows my and my partner's disability status represented by Line 8, "Blind/Disabled" being checked or unchecked for myself and my partner

>GIVEN I have submitted my state return
WHEN I view the XML
THEN it shows my and my partner's disability status

>GIVEN I have claimed the disability exemption
WHEN I view Line 13 and 30 (Total Exemption Amount) in the PDF and XML
THEN my disability exemption amount is reflected in this total sum

## Is PM acceptance required?
- Yes - don't merge until JIRA issue is accepted!

## What was done?
- Added intake question to check if primary or spouse are disabled
- Updated logic in calculator to update calculated exemption when primary and/or spouse are disabled
- Updated logic in XML submission builder to check Line 8 checkboxes if primary and/or spouse are disabled

## How to test?
  - Unit/integration/local testing
  - Created minimal import, validated that disabled question appears, it is required, it shows up on the review screen, and that it populates the XML and the PDF correctly when yes/no is selected
  - Created married filing jointly import, validated that disabled questions appears for primary and spouse, both fields are required, it shows up on the review screen, and that it populates the XML and the PDF correctly when yes/no is selected
  - Created blind_primary import, validated that disabled question does not appear and that the XML and PDF are populated correctly
  - Created blind_primary_and_spouse import, validated that disabled question does not appear and that the XML and PDF are populated correctly.

- All testing done locally

## Screenshots 
<img width="786" alt="image" src="https://github.com/user-attachments/assets/6f17c4a0-ce43-4185-b6a2-0c8d295a6e7e">
